src/compiler/compile/nodes/Element.ts

```diff
@@ -351,6 +351,53 @@ export default class Element extends Node {
 					const match = fuzzymatch(value, aria_roles);
 					component.warn(attribute, compiler_warnings.a11y_unknown_role(value, match));
 				}
+
+				// no-redundant-roles
+				if (this.name === value) {
+					component.warn(attribute, compiler_warnings.a11y_no_redundant_roles(value));
+				}
+				if (this.name === 'nav' && value === 'navigation') {
+					component.warn(attribute, compiler_warnings.a11y_no_redundant_roles(value));
+				}
+
+				if (this.name === 'a' && value === 'link') {
+					component.warn(attribute, compiler_warnings.a11y_no_redundant_roles(value));
+				}
+
+				if (this.name === 'fieldset' && value === 'group') {
+					component.warn(attribute, compiler_warnings.a11y_no_redundant_roles(value));
+				}
+
+				if (this.name === 'ul' && value === 'list') {
+					component.warn(attribute, compiler_warnings.a11y_no_redundant_roles(value));
+				}
+
+				if (this.name === 'main' && value === 'main') {
+					component.warn(attribute, compiler_warnings.a11y_no_redundant_roles(value));
+				}
```

Comment:
I think this is already covered above with `this.name === value`

Reply:
Ah yes, thanks!